src/lib.rs

```diff
@@ -235,6 +262,18 @@ impl Surface {
         })
     }
 
+    pub fn resize(&mut self, width: u32, height: u32) {
+        self.surface_impl.resize(width, height);
+    }
+
+    pub fn buffer_mut(&mut self) -> &mut [u32] {
+        self.surface_impl.buffer_mut()
+    }
+
+    pub fn present(&mut self) {
+        self.surface_impl.present();
```

Comment:
On Wayland this won't work unless the buffer size is at least 1x1, meaning a resize is needed before presenting. (You can present undefined contents however and be fine)./